### ISSUE TITLE
False positive isset operations variables existence: do while

### ISSUE THREAD

**Issue by Bilge** [kalessil/phpinspectionsea#1495]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) |
| Language level | PHP 7.4

Current behaviour
Variable declared later in the loop is presumed not to ever exist in scope.

Expected behaviour
Variables declared later will exist in subsequent operations and therefore may exist.

```php
do {
    $p = [];

    foreach ($new ?? $p as $k => $v) {
        $p[$k] = $v;
    }

    $new = [];

    foreach ($p as $k => $v) {
        if ($k > 5) $new[$k] = $v;
    }
while ($new);
```

In this contrived example, an the inspection errors on `$new` on the line containing `$new ?? $p`.

Environment details
2019.3

**Comment by kalessil** [comment]
Thank you for reporting @Bilge, I'll take care of fixing this.
